src/Model/ModelManager.php

```diff
@@ -409,17 +409,16 @@ public function getIdentifierValues($entity)
             }
 
             $fieldType = $metadata->getTypeOfField($name);
-            $type = $fieldType && Type::hasType($fieldType) ? Type::getType($fieldType) : null;
-            if ($type) {
+            if ($this->hasFieldCustomType($fieldType)) {
+                $type = $this->getFieldCustomType($fieldType);
```

Comment:
When I look at the logic inside these function, I'm not sure we need to add private method for this.

Reply:
It's used in two places. I usually try to remove duplicates. This reduces the burden on the developers when they need to update "Type" class usage.

Reply:
Yes, I saw the duplicate.

But it's basically like a method 
```
private function foo($a) {
    return $this->bar($a);
}
```

Reply:
Right. Similar to findBy and findOneBy https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/560e9f039d63cf2732e4f6c428549d9ac67826c5/src/Model/ModelManager.php#L285-L293

Reply:
But these one are public, and required by the interface.